src/Message/ItemDataTrait.php

```diff
@@ -23,10 +21,10 @@ public function getItemData(ItemBag $items)
                 'type' => $item->getType(),
                 'name' => $item->getName(),
                 'quantity' => $item->getQuantity(),
-                'tax_rate' => (int) ($item->getTaxRate() * 100),
-                'total_amount' => $this->toCurrencyMinorUnits($item->getQuantity() * $item->getPrice()),
-                'total_tax_amount' => $this->toCurrencyMinorUnits($item->getTotalTaxAmount()),
-                'unit_price' => $this->toCurrencyMinorUnits($item->getPrice()),
+                'tax_rate' => (int) $item->getTaxRate(),
+                'total_amount' => $item->getQuantity() * $item->getPrice(),
+                'total_tax_amount' => (int) $item->getTotalTaxAmount(),
```

Comment:
`getTaxRate` and `getTotalTaxAmount` return floats, so can not just be cast to int like this. I suggest restoring the `toCurrencyMinorUnits`, of perhaps some alternative trait which can convert a float/string to a `Money` object so it can be used both here and in `AbstractRequest::getAmount`